nano/lib/tomlconfig.cpp

```diff
@@ -186,6 +186,41 @@ void nano::tomlconfig::erase_default_values (tomlconfig & defaults_a)
 	erase_defaults (defaults_l.get_tree (), self.get_tree (), get_tree ());
 }
 
+// Merges two TOML configurations and commenting values that are identical
+std::string nano::tomlconfig::merge_defaults (nano::tomlconfig & current_config, nano::tomlconfig & default_config)
+{
+	auto default_tree = default_config.get_tree ();
+	auto current_tree = current_config.get_tree ();
+
+	debug_assert (default_tree != nullptr && current_tree != nullptr);
+
+	// Serialize both configs to commented strings
+	std::string defaults_str = default_config.to_string (true);
+	std::string current_str = current_config.to_string (true);
+
+	// Read both configs line by line
+	std::istringstream stream_defaults (defaults_str);
+	std::istringstream stream_current (current_str);
+	std::string line_defaults, line_current, result;
+
+	while (std::getline (stream_defaults, line_defaults) && std::getline (stream_current, line_current))
```

Comment:
This loop will only work correctly if the defaults and current lists are sorted and have the same number of elements. Is this guranteed to happen. If yes, how do you know?

Reply:
The current list and default list will always follow the exact same layout. This is because we use the serialize_toml function to create the configs. That function will create all the tree nodes, config items, and docs based on the input

Reply:
You can either remove it or check that it is "node".